syntax = "proto3";

service ClinicService {
  // Pacientes
  rpc CreatePatient (Patient) returns (Response);
  rpc GetPatient (PatientId) returns (Patient);
  rpc ListPatients (Empty) returns (PatientList);
  rpc UpdatePatient (Patient) returns (Response);
  rpc DeletePatient (PatientId) returns (Response);

  // Doctores
  rpc CreateDoctor (Doctor) returns (Response);
  rpc GetDoctor (DoctorId) returns (Doctor);
  rpc ListDoctors (Empty) returns (DoctorList);
  rpc UpdateDoctor (Doctor) returns (Response);
  rpc DeleteDoctor (DoctorId) returns (Response);

  // Citas
  rpc CreateAppointment (Appointment) returns (Response);
  rpc GetAppointment (AppointmentId) returns (Appointment);
  rpc ListAppointments (Empty) returns (AppointmentList);
  rpc UpdateAppointment (Appointment) returns (Response);
  rpc DeleteAppointment (AppointmentId) returns (Response);
}

message Empty {}

message Patient {
  int32 id = 1;
  string name = 2;
  int32 age = 3;
  string contact = 4;
}

message Doctor {
  int32 id = 1;
  string name = 2;
  string specialty = 3;
}

message Appointment {
  int32 id = 1;
  int32 patient_id = 2;
  int32 doctor_id = 3;
  string date = 4;
  string reason = 5;
}

message PatientId { int32 id = 1; }
message DoctorId { int32 id = 1; }
message AppointmentId { int32 id = 1; }

message PatientList {
  repeated Patient patients = 1;
}

message DoctorList {
  repeated Doctor doctors = 1;
}

message AppointmentList {
  repeated Appointment appointments = 1;
}

message Response {
  bool success = 1;
  string message = 2;
  int32 id = 3;
}